firedrake/mg/embedded.py

```diff
@@ -93,14 +93,17 @@ def cache(self, V):
         except KeyError:
             return self.caches.setdefault(key, TransferManager.Cache(*key))
 
+    def cache_key(self, V):
+        return (V.dim(), V.boundary_set)
+
```

Comment:
As discussed in the meeting, can we not put `V.boundary_set` in `key` (line 90)?